https://issues.redhat.com/browse/ACM-17011 (#8105)
open-cluster-management/rhacm-docs: global_hub/global_hub_requirements.adoc

[#global-hub-requirements]
= {global-hub} requirements

Learn about the {global-hub} requirements for installing and networking, and review the components and environments that are supported by {global-hub}.

> 

Reviewer: we don't need this:
{acm} must be installed and configured.

That is prereq info for a task.

Reviewer: This is not a task.

[#global-hub-requirements]
= {global-hub} requirements

> See the following general {global-hub} requirements:

Author: Doc reviewers: 

please let me know what you think of this restructure to remove prereqs in this concept doc.

[#global-hub-requirements]
= {global-hub} requirements

Learn about the {global-hub} requirements for installing and networking, and review the components and environments that are supported by {global-hub}. See the following requirements: 

* <<gh-prerequisites,Prerequisites>>
* <<gh-network-requirements,Networking requirements>>
* <<gh-supported-components,Supported components>>

[#gh-prerequisites]

> == Prerequisites

Reviewer: As I mentioned, since this is not a task, we prob should not have prereqs here... these are prereqs to the install task.

> To install {global-hub}, see the following prerequisites:

Reviewer: Line 13, this is not the install task. Why do we have this here?

Reviewer: I think you can keep the bullet points, but remove the header. What we may run in to is that is typical of a task and this is not a task.

* *Required access:* Cluster administrator
* *{ocp-short} Dedicated environment required access:* You must have `cluster-admin` permissions. By default `dedicated-admin` role does not have the required permissions to create namespaces in the {ocp-short} Dedicated environment. 
* {acm} must be installed and configured.

[#gh-network-requirements]
== Networking requirements

See the following networking requirements:

> * The managed hub is also a managed cluster of {global-hub} in {acm-short}. You must configure the network component in {acm-short}. For {acm-short} networking details, see link:../networking/networking_intro.adoc#networking[Networking]. 

Reviewer: Instances like this need to be reduced and turned around -- please scan the file for things like _of the_ which is placing the noun's adjective/modifier at the end of the sentence. 

Read this outloud:

**The managed hub is also a managed cluster of {global-hub} in {acm-short}.**

This is confusing. What are we trying to say here? And always use "hub cluster." Try:

The managed hub cluster is also a {global-hub} managed cluster in {acm-short}?

And should this be "local-cluster"? (managed hub cluster)

Reviewer: The changes to the table look good. Just remember to code the elements within the table, such as the pod name.

[#global-hub-requirements]

> = Global hub requirements

Author: Doc reviewers: I can use the conref for the title, but then would need to change all links to it. Didn't know how much we wanted to play around with xrefs and links at this moment.

Reviewer: Yes, we normally use conrefs in the title...you can see how big of a job to do that with a search in VS code and maybe save that for later. F and R is easiest in VS Code--moving to another editor, it may not be as easy.

> Learn about the {global-hub} requirements for installing and networking, and review the components and environments that are supported by {global-hub}. Complete the following sections: 

Author: Doc reviewers: I know this is self referential, but for usability and accessibility, I think we should introduce the bullet list of sections for the user to complete. Thoughts?

Reviewer: I think this is not the "self-referential" that we try to avoid. We want to avoid:

Use this process. Run this command: Follow this procedure...

As much as possible...I think this is fine.

One thing we need to chat about though is the fact that this is a concept, but we are telling them to complete something. We need to either make this a procedure, or take out that language that there is a task within. This will be even more strict with Dita, if it is similar to what I learned, but it is also the modular concept.

Reviewer: Since this is tech debt. I would use this to change things around a bit so it is not a blend of a task and concept. We have some examples of requirements topics that may help. It seems we have prereqs to install here.

To install Global hub, you need the following requirements:

* *Required access:* Cluster administrator
//this is normally at the beginning of a procedure and all of this as prereqs.

* *{ocp-short} Dedicated environment required access:* You must have `cluster-admin` permissions. By default `dedicated-admin` role does not have the required permissions to create namespaces in the {ocp-short} Dedicated environment. 
//same
* {acm} must be installed and configured.
//same

[#gh-network-requirements]
== Networking requirements

See the following networking requirements:

* The managed hub is also a managed cluster of {global-hub} in {acm-short}. You must configure the network component in {acm-short} is necessary. For {acm-short} networking details, see link:../networking/networking_intro.adoc#networking[Networking]. 

* To review the {global-hub} network information, see the following table:

+
|===
| Direction | Protocol | Connection | Port that is specified | Source address |	Destination address |
    

> | Inbound from browser of the user | HTTPS | The user needs to access the Grafana dashboard | 443 | Browser of the user | IP address of Grafana route |

Reviewer: Inbound from browser **of the user**? We say "of the user" multiple times here, and that is likely due to the different languages of the engineer and writer, but we should always simplify and change that from the drafts. 

We typically use conversational tone; I would try to simplify this:

Inbound from the browser| HTTPS | Accesses the Grafana dashboard | Browser | Grafana route IP address

(removing the "of the" language where you can)

And will need to be mindful of that in the entire book bc it comes to us like that in drafts sometimes.

> | Outbound to Kafka Cluster | HTTPS | The {global-hub} manager needs to get data from Kafka cluster | 443 | multicluster-global-hub-manager-xxx pod | Kafka route host |

Reviewer: simplify:
The {global-hub} manager needs to get data from Kafka cluster

receives data? May be more concise here.

> | Outbound to PostgreSQL database | HTTPS | The {global-hub} needs to persist data to PostgreSQL database | 443 | multicluster-global-hub-manager-xxx pod | IP address of the PostgreSQL database |

Reviewer: Same suggestion to simplify line 38 if possible. Other cells are simplified and then some are not as concise in the table.

|===
+

* To review the managed hub network information, see the following table: 

+
|===
| Direction | Protocol | Connection | Port that is specified | Source address |	Destination address |
    

> | Outbound to Kafka Cluster | HTTPS | The {global-hub} agent needs to sync cluster info and policy info to Kafka cluster | 443 | multicluster-global-hub-agent pod | Kafka route host |

Reviewer: we don't write "info" so we need to spell that out and you can do that by reducing other words:

_The {global-hub} agent syncs cluster and policy information_

|===
+

> * For guidelines for sizing, see link:../install/cluster_size.adoc#sizing-your-cluster[Sizing your {acm-short} cluster].

Reviewer: ```suggestion
* For sizing guidelines, see link:../install/cluster_size.adoc#sizing-your-cluster[Sizing your {acm-short} cluster].
```

[#gh-supported-components]
== Supported components

> See the following components that {global-hub} supports: 

Reviewer: ```suggestion
See the following supported components: 
```

Reviewer: (they know they are reading a gh document)

> * Because they share an integrated console, the {global-hub} and {ocp-short} console share an integrated console, so they support the same browser. For information about supported browsers and versions, see link:https://docs.redhat.com/documentation/en-us/openshift_container_platform/4.17/html-single/web_console/index#web-console[Accessing the web console] in the {ocp} documentation

Reviewer: Reading this out loud, we say the same thing twice? and we use anthropomorphism:

Because they share an integrated console, the {global-hub} and {ocp-short} console share an integrated console, so they support the same browser. (???)

Why not just say

The integrated console for ocp and gh supports the same browser (??).  If we understand what the line is saying here, we can greatly reduce this. 

The OCP and ACM console is integrated and support the same browser. (Also a way to say that)

> * See following table for a list of platforms that are available for the supported {global-hub} cluster:

Reviewer: ```suggestion
* See the following table of supported platforms for the {global-hub} cluster:
```

Reviewer: we do need to make sure the tables are titled...I have not checked all of them.

|{acm-short} on IBM Z | Yes | Yes
|{acm-short} on IBM Power Systems | Yes | Yes
|===
+

> * {global-hub} supports middleware, including: Kafka, PostgreSQL, and Grafana. See the following table for a list of supported middleware and built-in versions:

Reviewer: ```suggestion
* {global-hub} supports middleware, such as Kafka, PostgreSQL, and Grafana. See the following table for a list of supported middleware and built-in versions:
```

Reviewer: including: 

Doesn't always orient the user that a list is following, because you just use a colon with no language. Good idea to avoid it.
